doc/advanced/content/executor_design.rst

```diff
@@ -374,15 +374,15 @@ large parts of the codebase would need to be refactored to support executors.
 
 The idea was to have a single base executor from which all executors would derive from. This would have allowed
 all the common code to be shared among all the executors. The base executor had a CRTP based design that
-was used to access the properties of the derived executor. It also allowed simplification in many areas, such as being able to reference any derived executor. Basic properties of all executors like
-copy constructors, overloaded equality operators.
-The CRTP mechanism had some restrictions in the sense that there was still a need for templates, and
-there were concerns regarding explicitly passing all the properties to the base class using CRTP as
-it was felt to be unnecessary. Having inheritance also introduces runtime polymorphism as the derived executor
-would override some of the base executors' method, and the call to the overridden function is resolved at run time.
-This is opposed to one of the design consideration i.e., have everything compile-time and avoid any overhead.
-The base executor didn't add a lot besides allowing code sharing of a few common functionalities and thus
-it was discarded.
+was used to access the properties of the derived executor. It also allowed simplifications in many areas, such as
+being able to reference any derived executor. Basic properties of all executors like
+copy constructors, overloaded equality operators. The CRTP mechanism had some restrictions in the sense that
+there was still a need for templates, and there were concerns regarding explicitly passing all the properties
+to the base class using CRTP as it was felt to be unnecessary. Having inheritance also introduces runtime polymorphism
+as the derived executor would override some of the base executors' method, and the call to the overridden function is
+resolved at run time. This is opposed to one of the design consideration i.e., have everything compile-time and avoid
+any overhead. The base executor didn't add a lot besides allowing code sharing of a few common functionalities and
```

Comment:
...and avoid runtime overhead. 


You still pay the price in compilation time. 
Something to watch in case you're feeling lazy at some point:
https://www.youtube.com/watch?v=rHIkrotSwcc

Reply:
> Something to watch in case you're feeling lazy at some point:

Links to a 1 hour info-dense video by Chandler. I wonder how Sergio relaxes 😆